File: UI/careconnect_admin/lib/models/requests/appointment_insert_request.dart
import 'package:json_annotation/json_annotation.dart';

part 'appointment_insert_request.g.dart';

@JsonSerializable()
class AppointmentInsertRequest {
  int userId;
  int employeeAvailabilityId;
  String appointmentType;
  int attendanceStatusId;
  DateTime date;
  String? description;
  String? note;

  AppointmentInsertRequest({
    required this.userId,
    required this.employeeAvailabilityId,
    required this.appointmentType,
    required this.attendanceStatusId,
    required this.date,
    this.description,
    this.note,
  });

  factory AppointmentInsertRequest.fromJson(Map<String, dynamic> json) =>
      _$AppointmentInsertRequestFromJson(json);

  Map<String, dynamic> toJson() => _$AppointmentInsertRequestToJson(this);
}

File: CareConnect/CareConnect.Models/Requests/AppointmentInsertRequest.cs
using CareConnect.Models.Responses;
using System;
using System.Collections.Generic;
using System.Text;
using System.Text.Json.Serialization;

namespace CareConnect.Models.Requests
{
    public class AppointmentInsertRequest
    {
        public int EmployeeAvailabilityId { get; set; }

        public int ClientId { get; set; }

        public int ChildId { get; set; }

        public string AppointmentType { get; set; } = null!;

        public int AttendanceStatusId { get; set; }

        public DateTime Date { get; set; }

        public string? Description { get; set; }

        public string? Note { get; set; }

        public string? PaymentIntentId { get; set; } = null; // null for free items

        [JsonIgnore]
        public DateTime ModifiedDate { get; set; } = DateTime.Now;
    }
}
